### PR TITLE
SpacingCheck: add "for", "while" and "switch" support, closes #46

### LINKED ISSUE #46: Check for if / switch / for / while spacing
There are (at least) two different styles you might want to enforce here:

`if (` / `switch (` / `for (` and `while (`

or

`if(` / `switch(` / `for(` and `while(`

I don't think there currently is a check that handles this.

### PR DESCRIPTION
**Pull request by Gama11**

### CODE REVIEW

File: src/checkstyle/checks/SpacingCheck.hx [hunk 1]
```diff
@@ -41,8 +48,19 @@ class SpacingCheck extends Check {
...
-				case EIf(econd, eif, eelse) if (spaceIfCondition):
-					if (econd.pos.min - e.pos.min < "if (".length) logPos('No space between if and condition', e.pos, Reflect.field(SeverityLevel, severity));
+				case EIf(econd, _, _) if (spaceIfCondition):
+					checkSpaceBetweenExpressions('if', e, econd);
+				case EFor(it, _) if (spaceForLoop):
+					checkSpaceBetweenExpressions('for', e, it);
+				case EWhile(econd, _, true) if (spaceWhileLoop):
+					checkSpaceBetweenExpressions('while', e, econd);
+				case ESwitch(eswitch, _, _) if (spaceSwitchCase):
+					var prevExprUntilSwitch = checker.file.content.substring(lastExpr.pos.min, eswitch.pos.min + 1);
```
Comment (Gama11) [inline src/checkstyle/checks/SpacingCheck.hx:58]: This is a bit messy, but I couldn't get it to work properly with just the `ESwitch` positions like with `if`, `for` and `while`.

File: src/checkstyle/checks/SpacingCheck.hx [hunk 2]
```diff
@@ -41,8 +48,19 @@ class SpacingCheck extends Check {
 					if (post) dist = e.pos.max - e2.pos.max;
 					else dist = e2.pos.min - e.pos.min;
 					if (dist > unopSize(uo)) logPos('Space around ${unopString(uo)}', e.pos, Reflect.field(SeverityLevel, severity));
-				case EIf(econd, eif, eelse) if (spaceIfCondition):
-					if (econd.pos.min - e.pos.min < "if (".length) logPos('No space between if and condition', e.pos, Reflect.field(SeverityLevel, severity));
+				case EIf(econd, _, _) if (spaceIfCondition):
+					checkSpaceBetweenExpressions('if', e, econd);
+				case EFor(it, _) if (spaceForLoop):
+					checkSpaceBetweenExpressions('for', e, it);
+				case EWhile(econd, _, true) if (spaceWhileLoop):
```
Comment (Gama11) [inline src/checkstyle/checks/SpacingCheck.hx:55]: Only handling regular `while` loops here for now, `do...while` is a bit trickier.

**Comment by Gama11** [comment]
Eventually this check should probably support the inverse as well (enforcing there are no spaces).

**Comment by adireddy** [comment]
:+1:

**Pull request merged**